docs/architecture/overview.md

```diff
@@ -47,8 +49,24 @@ The check before a transaction can e.g. programmatically check all of the parame
 > ⚠️ IMPORTANT: Since a guard has full power to block Safe transaction execution,
 > a broken guard can cause a denial of service for the Safe. Make sure to carefully audit the guard code and design recovery mechanisms.
 
+To learn more about guards, please refer to the [guards](./guards.md) documentation.
+
+#### Libraries
+
+Libraries implement custom execution flows that are optimized towards specific protocols or use cases. To use the libraries they are called using `delegatecall`, which executes the library logic via the Safe contract.
```

Comment:
```suggestion
Libraries implement custom execution flows that are optimized towards specific protocols or use cases. To use the libraries they are called using [delegatecall](https://docs.soliditylang.org/en/latest/introduction-to-smart-contracts.html#delegatecall-and-libraries), which executes the library logic via the Safe contract.
```

Reply:
I would not link this here and rather explain it in the detailed documentation on libraries.